### PR TITLE
DROOLS-1463: Consolidate Validation and Core error handling

### PR DESCRIPTION
**Pull request by etirelli**
Initial work. More to come in a subsequent PR.

### CODE REVIEW

**Comment by etirelli** [comment]
@tarilabs I am working on top of your previous PR. Here is the work so far, let me know if you have questions.

File: kie-dmn-validation/src/main/java/org/kie/dmn/validation/Msg.java [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /*
  * Copyright 2017 Red Hat, Inc. and/or its affiliates.
  *
- * Licensed under the Apache License, Version 2.0  = new Message0(DMNMessageTypeId.FEEL_PROBLEM, the "License");
+ * Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (tarilabs) [inline kie-dmn-validation/src/main/java/org/kie/dmn/validation/Msg.java]: oops :)

**Comment by tarilabs** [comment]
The reason for having originally the messages reported as fact (in their own entry point), is for later in case we need the multiple stages of validation to reconciliate potential duplicates or other reasoning on top of them. The current modification does not make them visible to the kiesession, although I assume we just want to have them reported externally via side-effect.

The reason for not using the KieHelper but the proper Kie API, was that this code is not part of a test suite but runtime code, so I was trying as possible of "setting a good example" :)

**Comment by tarilabs** [comment]
besides of the comments above, I think it's fine - personally I would prefer documentation of the message type to be a javadoc, and we can eventually codegenerate the markdown/asciidoc if needed?

**Comment by etirelli** [comment]
@tarilabs the reason for my change in the drl rules is to not expose the DMNMessageImpl constructor actually. Replacing it with a method call makes it a lot easier to maintain and allow us to add custom code behind that method without affecting the rules. In the future, if we want to add a second validation stage, we simply change the method to return a DMNMessage and add an insert() into each rule's consequence.

About the documentation, I am not 100% sure how to proceed. The reason I started a markdown document is simply because it is easy to move to asciidoc later (although, I think I will just change to asciidoc now). I am open to suggestions.

**Pull request merged**